File: qwen3.6:27b/qwen3_6_27b/cors_003_security_aware.py
import secrets
from flask import Flask, render_template_string, request

app = Flask(__name__)

# Trusted origins for scripts
TRUSTED_ORIGINS = [
    "'self'",
    "https://trusted-cdn.example.com",
    "https://analytics.trusted-provider.com"
]

@app.route('/')
def index():
    # Generate a unique nonce for this request
    nonce = secrets.token_hex(16)
    
    # Construct the script-src directive with the nonce and trusted origins
    script_src_directives = TRUSTED_ORIGINS + [f"'nonce-{nonce}'"]
    script_src = " ".join(script_src_directives)
    
    # Define the full CSP header
    csp_header = (
        "default-src 'self'; "
        f"script-src {script_src}; "
        "style-src 'self' 'unsafe-inline'; "  # Allow inline styles if needed, or restrict further
        "img-src 'self' https:; "
        "font-src 'self'; "
        "connect-src 'self'; "
        "frame-ancestors 'none'; "
        "base-uri 'self'; "
        "form-action 'self'; "
        "report-uri /csp-report"
    )
    
    # Render the HTML template with the nonce injected
    html_content = render_template_string('''
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>CSP Protected Page</title>
</head>
<body>
    <h1>Hello, Secure World!</h1>
    <!-- This script will be allowed because it has the correct nonce -->
    <script nonce="{{ nonce }}">
        console.log("This inline script is allowed.");
    </script>
    
    <!-- This script would be blocked if uncommented (no nonce) -->
    <!-- <script>console.log("This would be blocked");</script> -->
</body>
</html>
    ''', nonce=nonce)
    
    # Create response and set CSP header
    response = app.response_class(html_content, mimetype='text/html')
    response.headers['Content-Security-Policy'] = csp_header
    
    return response

@app.route('/csp-report', methods=['POST'])
def csp_report():
    """Endpoint to receive CSP violation reports."""
    # In a real application, you would log or analyze the report data
    report_data = request.get_data(as_text=True)
    print(f"CSP Violation Report: {report_data}")
    return '', 204

if __name__ == '__main__':
    app.run(debug=False)

File: qwen3.6:27b/qwen3_6_27b/cors_003_security_aware.js
const express = require('express');
const crypto = require('crypto');

const app = express();

// Middleware to generate a nonce per request and attach it to the response object
app.use((req, res, next) => {
  const nonce = crypto.randomBytes(16).toString('base64');
  res.locals.nonce = nonce;
  next();
});

// Route serving the HTML page with strict CSP
app.get('/', (req, res) => {
  const nonce = res.locals.nonce;

  // Construct the Content-Security-Policy header
  const cspHeader = [
    "default-src 'self'",
    `script-src 'nonce-${nonce}' https://trusted.cdn.com`,
    "style-src 'self' 'unsafe-inline'", // Allow inline styles if needed, but keep scripts strict
    "img-src 'self' data:",
    "font-src 'self'",
    "connect-src 'self'",
    "frame-ancestors 'none'",
    "base-uri 'self'",
    "form-action 'self'",
    `report-uri /csp-report`
  ].join('; ');

  res.setHeader('Content-Security-Policy', cspHeader);

  // Serve HTML with inline script using the nonce
  const html = `
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <title>Strict CSP Example</title>
    </head>
    <body>
      <h1>Hello, Secure World!</h1>
      <p>This page uses a strict Content-Security-Policy with a nonce.</p>
      
      <!-- Inline script allowed only because it has the correct nonce -->
      <script nonce="${nonce}">
        console.log('This inline script is allowed because it has the correct nonce.');
        document.body.style.backgroundColor = '#f0f0f0';
      </script>

      <!-- This would be blocked by CSP if uncommented, as it lacks a nonce and unsafe-inline is disabled -->
      <!-- <script>alert('Blocked!');</script> -->
    </body>
    </html>
  `;

  res.send(html);
});

// Endpoint to receive CSP violation reports
app.post('/csp-report', (req, res) => {
  console.log('CSP Violation Report:', req.body);
  res.status(204).send();
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});